### DISCUSSION TITLE
Would there be any way to stack nvim-tree with another plugin in the same sidebar column?

### DISCUSSION THREAD

**Discussion by mrjones2014** [nvim-tree/nvim-tree.lua#2173] [Q&A]
I've been using [nvim-ide](https://github.com/ldelossa/nvim-ide) for several months now, the main reason being that I like the ability to have multiple components stacked in the same column/sidebar. However pretty much the only components I actually use from it are the bufferlist and file tree, and sometimes symbols outline (see screenshot, this is how I usually have it when I toggle it open).

![image](https://user-images.githubusercontent.com/8648891/235226104-7226f658-f2c5-4ab8-9f8b-c97c44e5f93a.png)

Would there be any way to do something like this with `nvim-tree`? I could use [aerial.nvim](https://github.com/stevearc/aerial.nvim) for the symbols outline and it'd be trivial enough to write my own bufferlist component.

**Comment by alex-courtis** [comment]
aerial and nvim-tree at the left would work quite nicely. I might use that.

RE buffer list you can use `B` `api.tree.toggle_no_buffer_filter`

This might be possible without any nvim-tree or aerial changes. Experimenting.

**Reply by alex-courtis** [reply]
Didn't have much luck:
`wincmd x`, `r` etc can only be used inside a row or column - no way to get the two windows into a column or row
opening a new window and switching to the nvim-tree / aerial buffer upset them

**Reply by alex-courtis** [reply]
We could do something like:
```lua
  api.tree.open({
    relative = {
      ft = "aerial",
      position = "above",
    },
  })
```

We'd need something similar from aeiral for this to be workable.

**Reply by mrjones2014** [reply]
If aerial had an option to just open in the current window, we could use that, right?

**Reply by alex-courtis** [reply]
Yes, that might work. nvim-tree can open in current window.

The user could create a function something like:
```
:topleft vert new  -- record winid $bottom
:horiz new -- record winid $top
:$top wincmd w
nvim_tree_api.tree.open({current_window} = true, })
:$bottom wincmd w
aerial_api.open({current_window} = true, })
```

Repositioning might be tricky, but it's a start.

I'd be grateful if you could open an issue / PR for aerial @mrjones2014

**Reply by mrjones2014** [reply]
https://github.com/stevearc/aerial.nvim/issues/248

**Reply by alex-courtis** [reply]
Thank you

**Reply by mrjones2014** [reply]
Aerial added `aerial.open_in_win(target_win, source_win)` and I've got a hackjob somewhat working like this:

```lua
local width = 35

return {
  'nvim-tree/nvim-tree.lua',
  cmd = 'Tree',
  dependencies = {
    'stevearc/aerial.nvim',
    config = function()
      require('aerial').setup({
        layout = {
          max_width = { width },
          min_width = { width },
          placement = 'edge',
        },
        attach_mode = 'global',
        keymaps = {
          ['<C-j>'] = false,
          ['<C-k>'] = false,
          ['<C-y>'] = 'actions.up_and_scroll',
          ['<C-e>'] = 'actions.down_and_scroll',
        },
      })
    end,
  },
  config = function()
    require('legendary').command({
      'Tree',
      function()
        local curwin = vim.api.nvim_get_current_win()
        local a = require('aerial')
        local tree = require('nvim-tree.api').tree
        tree.open({ path = vim.loop.cwd() })
        local orig = vim.opt.splitbelow
        vim.opt.splitbelow = false
        vim.cmd(('%ssp'):format(width))
        vim.cmd('wincmd j')
        vim.opt.splitbelow = orig
        vim.schedule(function()
          a.open_in_win(0, curwin)
          vim.api.nvim_win_set_width(0, width)
          -- put cursor focus back where it was
          vim.api.nvim_set_current_win(curwin)
        end)
      end,
    })
    require('nvim-tree').setup({ view = { side = 'right' } })
  end,
}
```

Now I just need to create a bufferlist component for myself.

> RE buffer list you can use B api.tree.toggle_no_buffer_filter

I'd like to have _both_ a file tree and a bufferlist though, so I'd need two instances of nvim-tree to do that, which I'm not sure is possible?

**Reply by mrjones2014** [reply]
Hmm, with this code, `nvim-tree` seems to lose track of its buffer, and `tree.close()` closes the Aerial window instead.

**Reply by alex-courtis** [reply]
Wow... thanks for the quick feature @stevearc 

Implementing the same API as aerial in nvim-tree seems the best course of action: allow full control for the user instead of relative positioning.

**Reply by alex-courtis** [reply]
#2189
